dowhy/causal_identifier/identify_effect.py

```diff
@@ -0,0 +1,190 @@
+import copy
+from enum import Enum
+from typing import List, Protocol, Union
+
+import sympy as sp
+
+from dowhy.causal_graph import CausalGraph
+from dowhy.utils.api import parse_state
+
+
+class CausalIdentifierEstimandType(Enum):
+    # Average total effect
+    NONPARAMETRIC_ATE = "nonparametric-ate"
+    # Natural direct effect
+    NONPARAMETRIC_NDE = "nonparametric-nde"
+    # Natural indirect effect
+    NONPARAMETRIC_NIE = "nonparametric-nie"
+    # Controlled direct effect
+    NONPARAMETRIC_CDE = "nonparametric-cde"
+
+
+class CausalIdentifier(Protocol):
+    """
+    Protocol to define a CausalIdentifier, all CausalIdentifiers must conform to at least this list of methods.
+    """
+
+    def identify_effect(
+        self, graph: CausalGraph, treatment_name: Union[str, List[str]], outcome_name: Union[str, List[str]], **kwargs
+    ):
+        """Identify the causal effect to be estimated based on a CausalGraph
+        :param graph: CausalGraph to be analyzed
+        :param treatment_name: name of the treatment
+        :param outcome_name: name of the outcome
+        :param **kwargs: Additional parameters required by the identify_effect of a specific CausalIdentifier
```

Comment:
I think we will also need observed_node_names here once we switch to a networkx object for causal graph. But that's for later.

Reply:
Ok, I'll resolve this conversation as it sounds like there's no current actions needed.